### PR TITLE
Increase PSK_MAX_PSK_LEN from 256 to 512

### PR DESCRIPTION
**Pull request by rutgerhendriks**
Increase PSK_MAX_PSK_LEN from 256 to 512, so OpenSSL can connect to a device that only accepts 512 byte psk's

### CODE REVIEW

**Comment by rutgerhendriks** [comment]
As mentioned in #12771 the rfc recommends longer identity and psk's.

**Comment by mattcaswell** [comment]
Closse/reopen to kick CLA bot

**Review by mattcaswell — APPROVED** [review]
Approved, subject to a rebase. Needs second review.

**Comment by slontis** [comment]
Needs a rebase..

**Comment by mattcaswell** [comment]
This PR has merge commits which are not allowed. They need to be removed and the PR rebased against the current master instead.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
The force push was a clean rebase so no re-review is needed.

**Comment by t8m** [comment]
Merged to master. Thank you for the contribution.

**Pull request closed without merging**
